window_drag_resize fullscreen window crashing

### Description

The full-screen window was removed by sliding out to the left off the screen, then stretched from the bottom right corner, and then it crashed. like #3245 

window: PhysicalPosition { x: 0, y: 0 }, pointer: 2614 0
[2024-03-18T02:01:43Z WARN  wgpu_hal::vulkan] Suboptimal present of frame 0
window: PhysicalPosition { x: 0, y: 163 }, pointer: 2823 29
window: PhysicalPosition { x: 0, y: 143 }, pointer: 2801 22
window: PhysicalPosition { x: 0, y: 133 }, pointer: 2938 23
window: PhysicalPosition { x: -568, y: 135 }, pointer: 3836 1922
thread 'main' panicked at src/platform_impl/linux/x11/window.rs:1748:21:
attempt to add with overflow
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

### OS and window mananger

OS: Manjaro Linux x86_64
Resolution: 3840x2160
DE: Plasma 5.27.11 
WM: KWin

### Winit version

0.29.11

> Ah, it's issue, see the linked PR with the proposed fix.

Ah. And I actually came up with the exact same patch as the PR. No PR from me then.

>  I don't understand why we're sending u32, where the data is actually i32?

Well... I guess the reason for this is "because C" and "implicit conversions". The [spec for `_NET_WM_MOVERESIZE`](https://specifications.freedesktop.org/wm-spec/1.4/ar01s04.html) doesn't say much about data types. Or at least I couldn't find it. Anyway, these fields should be set to:

> x_root and y_root MUST indicate the position of the button press with respect to the root window

I think this means these values cannot be negative. The root window always has its top left corner at (0, 0) and I think the mouse cursor can never be beyond that. So, in a sense, the data is not really `i32`, but really is `u32`. Just the intermediate computation isn't.

The other reason why this sends an unsigned value is that this is how `ClientMessage` requests are defined in X11 and thus also in x11rb. This request always contains unsigned values. In the words of the compiler:

```
error[E0277]: the trait bound `x11rb::protocol::xproto::ClientMessageData: From<[i32; 5]>` is not satisfied
    --> src/platform_impl/linux/x11/window.rs:1791:17
     |
1783 |               .send_client_msg(
     |                --------------- required by a bound introduced by this call
...
1791 | /                 [
1792 | |                     (window.x + xinput_fp1616_to_float(pointer.win_x) as i32),
1793 | |                     (window.y + xinput_fp1616_to_float(pointer.win_y) as i32),
1794 | |                     action.try_into().unwrap(),
1795 | |                     1, // Button 1
1796 | |                     1,
1797 | |                 ],
     | |_________________^ the trait `From<[i32; 5]>` is not implemented for `x11rb::protocol::xproto::ClientMessageData`
     |
     = help: the following other types implement trait `From<T>`:
               <x11rb::protocol::xproto::ClientMessageData as From<[u16; 10]>>
               <x11rb::protocol::xproto::ClientMessageData as From<[u32; 5]>>
               <x11rb::protocol::xproto::ClientMessageData as From<[u8; 20]>>
     = note: required for `[i32; 5]` to implement `Into<x11rb::protocol::xproto::ClientMessageData>`
note: required by a bound in `client_msg::<impl xdisplay::XConnection>::send_client_msg`
    --> src/platform_impl/linux/x11/util/client_msg.rs:11:20
     |
5    |     pub fn send_client_msg(
     |            --------------- required by a bound in this associated function
...
11   |         data: impl Into<xproto::ClientMessageData>,
     |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `client_msg::<impl XConnection>::send_client_msg`

```

--- a/src/platform_impl/linux/x11/window.rs
+++ b/src/platform_impl/linux/x11/window.rs
@@ -1789,8 +1789,8 @@ impl UnownedWindow {
                         | xproto::EventMask::SUBSTRUCTURE_NOTIFY,
                 ),
                 [
-                    (window.x as u32 + xinput_fp1616_to_float(pointer.win_x) as u32),
-                    (window.y as u32 + xinput_fp1616_to_float(pointer.win_y) as u32),
+                    (window.x + xinput_fp1616_to_float(pointer.win_x) as i32) as u32,
+                    (window.y + xinput_fp1616_to_float(pointer.win_y) as i32) as u32,
                     action.try_into().unwrap(),
                     1, // Button 1
                     1,

